app/models/request/sample_compound_aliquot_transfer.rb

```diff
@@ -50,13 +50,15 @@ def aliquots_by_tags_combination
     asset.aliquots.group_by(&:tags_combination)
   end
 
+  # For a group of source aliquots, find or create a compound sample containing the component samples
+  # Assign the compound sample to the target asset
   def transfer_into_compound_sample_aliquot(source_aliquots)
     compound_aliquot = CompoundAliquot.new(request: self, source_aliquots: source_aliquots)
     unless compound_aliquot.valid?
       raise Request::SampleCompoundAliquotTransfer::Error, compound_aliquot.errors.full_messages
     end
 
-    compound_aliquot.create_compound_sample
+    compound_aliquot.find_or_create_compound_sample(source_aliquots)
```

Comment:
I don't think you have to pass in `source_aliquots` here, because you already passed it in on line 56?